### PR TITLE
Pivotal #Id: 172415684: Store user direct submission into user folder

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/story/show/172415684

- fixed attribute support
- separated submission from data controllers

### CODE REVIEW

File: submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/handlers/SubmitWebHandler.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import ebi.ac.uk.model.Submission
+import ebi.ac.uk.model.SubmissionMethod.FILE
+import ebi.ac.uk.model.SubmissionMethod.PAGE_TAB
+import ebi.ac.uk.model.extensions.rootPath
+import ebi.ac.uk.security.integration.model.api.SecurityUser
+import java.io.File
+
+private const val DIRECT_UPLOAD_PATH = "direct-uploads"
+
+class SubmitWebHandler(
```
Comment (jhoanmanuelms) [inline submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/handlers/SubmitWebHandler.kt:22]: Why did you change the name? I think `SubmissionwebHandler` is better

File: submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/handlers/SubmitWebHandler.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        return submissionService.submit(SubmissionRequest(submission, request.user, source, FILE, request.fileMode))
+    }
+
+    private fun sources(user: SecurityUser, submission: Submission, files: List<File> = emptyList()): FilesSource {
+        val request = RequestSources(user, files, submission.rootPath, subFolder(submission.accNo))
+        return sourceGenerator.submissionSources(request)
+    }
+
+    private fun withAttributes(submission: Submission, attrs: Map<String, String>): Submission {
+        attrs.forEach { submission[it.key] = it.value }
```
Comment (jhoanmanuelms) [inline submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/handlers/SubmitWebHandler.kt:49]: I think this one-liner would work? `submission.also { attrs.forEach { submission[it.key] = it.value } }`

Comment (Juan-EBI) [reply submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/handlers/SubmitWebHandler.kt:49]: do not like in this case as we still need to use consturctor parameter submission as []`notation does not work here
